### PR TITLE
Feature/bingo elastic reactions

### PR DESCRIPTION
**Pull request by khyurri**
This PR adds reaction support  bingo-elastic implementation (Java and Python)

What changed:

1. Instance of IndigoRecord class cannot be saved to Elasticsearch. Developer must create instance of IndigoRecordMolecule or IndigoRecordReaction for saving to Elasticsearch.
2. Index bingo renamed to bingo-molecule. Added index bingo-reaction
3. Developer must provide index when creates an instance of ElasticRepository

### CODE REVIEW

File: api/plugins/bingo-elastic/java/src/main/java/com/epam/indigo/elastic/ElasticRepository.java [hunk 1]
```diff
@@ -219,8 +219,13 @@ public ElasticRepositoryBuilder() {
...
 
-        public ElasticRepositoryBuilder<T> withIndexName(String indexName) {
-            operations.add(repo -> repo.indexName = indexName);
+        public ElasticRepositoryBuilder<T> withIndexName(IndexName indexName) {
+            operations.add(repo -> repo.indexName = indexName.toString());
+            return this;
+        }
+
+        public ElasticRepositoryBuilder<T> withIndexName(IndexName.Index indexName) {
+            operations.add(repo -> repo.indexName = (new IndexName(indexName)).toString());
```
Comment (MysterionRise) [inline api/plugins/bingo-elastic/java/src/main/java/com/epam/indigo/elastic/ElasticRepository.java:228]: why you need to copy constructor here?

File: api/plugins/bingo-elastic/java/src/main/java/com/epam/indigo/elastic/IndexName.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        }
+        if (indigoRecord instanceof IndigoRecordReaction) {
+            return new IndexName(Index.BINGO_REACTION);
+        }
+        throw new BingoElasticException("Unknown IndigoRecord type " + indigoRecord.getClass());
+    }
+
+    @Override
+    public String toString() {
+        return this.index.toString().toLowerCase();
```
Comment (MysterionRise) [inline api/plugins/bingo-elastic/java/src/main/java/com/epam/indigo/elastic/IndexName.java:33]: if you want to do to string - you may do custom toString for Enum to provide name friendly representation.

File: api/plugins/bingo-elastic/java/src/test/java/com/epam/indigo/elastic/ElasticRepositoryTest.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.epam.indigo.elastic;
+
+import org.junit.Test;
+
+public class ElasticRepositoryTest {
+
+    @Test
+    public void testIndex() {
+
```
Comment (MysterionRise) [inline api/plugins/bingo-elastic/java/src/test/java/com/epam/indigo/elastic/ElasticRepositoryTest.java:9]: Empty test?

**Review by MysterionRise — CHANGES_REQUESTED** [review]
Overall, I would like to discuss those changes and ultimately I'm keen to have it differently than enums, but should be open for discussion

File: api/plugins/bingo-elastic/java/src/main/java/com/epam/indigo/elastic/IndexName.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+import com.epam.indigo.BingoElasticException;
+import com.epam.indigo.model.IndigoRecord;
+import com.epam.indigo.model.IndigoRecordMolecule;
+import com.epam.indigo.model.IndigoRecordReaction;
+
+public class IndexName {
+
+    public static final String BINGO_MOLECULE = "bingo_molecules";
+    public static final String BINGO_REACTION = "bingo_reactions";
```
Comment (MysterionRise) [inline api/plugins/bingo-elastic/java/src/main/java/com/epam/indigo/elastic/IndexName.java:11]: forgot to rename constants name to BINGO_MOLECULES

File: api/plugins/bingo-elastic/python/bingo_elastic/elastic.py [hunk 5]
```diff
@@ -1,32 +1,58 @@
...
+)
 from bingo_elastic.queries import BaseMatch, query_factory
 from bingo_elastic.utils import PostprocessType
 
 
-class ElasticRepository:
+class IndexName(Enum):
+
+    BINGO_MOLECULE = "bingo-molecules"
+    BINGO_REACTION = "bingo-reactions"
```
Comment (MysterionRise) [inline api/plugins/bingo-elastic/python/bingo_elastic/elastic.py:22]: same here

**Review by MysterionRise — APPROVED** [review]
In general, I'm good, will update some Java code with code from #251

File: api/plugins/bingo-elastic/java/src/test/java/com/epam/indigo/elastic/FullUsageMoleculeTest.java [hunk 6]
```diff
@@ -241,6 +258,15 @@ public void wildcardSearch() {
         } catch (Exception exception) {
             Assertions.fail("Exception happened during test " + exception.getMessage());
         }
+    }
 
+    @Test
+    @DisplayName("Testing TanimotoSimilarityMatch reaction")
+    public void reactionTanimoto() {
+        try {
+
```
Comment (MysterionRise) [inline api/plugins/bingo-elastic/java/src/test/java/com/epam/indigo/elastic/FullUsageMoleculeTest.java:267]: Empty test for now?

**Comment by khyurri** [comment]
Java and Python versions are ready for review

**Review by MysterionRise — APPROVED** [review]

**Review by mkviatkovskii — APPROVED** [review]

**Comment by mkviatkovskii** [comment]
I see some tests are still failing. We can merge the request after fixing that.

**Pull request merged**